### ISSUE TITLE
PCM file not found

### ISSUE THREAD

**Issue by belmakier** [root-project/root#9697]
[x] Checked for duplicates

### Describe the bug
Error message

`Error in <TCling::LoadPCM>: ROOT PCM <root_build_directory/bin/library_rdict.pcm> file does not exist`

is printed when a dynamic library is loaded in the ROOT prompt, even when the appropriate *_rdict.pcm file exists in the same directory as the library.

### Expected behavior
ROOT should search both the $LD_LIBRARY_PATH and the directory where the library is located for the *_rdict.pcm file.

### To Reproduce
Load a dynamic library with 
`gROOT->ProcessLine(".L path_to_library.dylib")`
or equivalent

### Setup
1. ROOT 6.27/01
2. macOS 11.4
3. built from source

**Comment by belmakier** [comment]
As far as I can tell, this is related to the following snippet from [here](https://github.com/root-project/cling/blob/1ebe5a8c91662665654af39aa0e73ef638343c42/lib/Interpreter/DynamicLibraryManagerSymbol.cpp#L348).

```
  // Handle path list items
  for (auto item : p) {
    if (item.startswith(".")) {
      if (item == "..") {
        size_t s = result.rfind(sep);
        if (s != llvm::StringRef::npos) result.resize(s);
        if (result.empty()) result = sep;
      }
      continue;  
    } else if (item == "~") {
      continue;
    }
```

The absolute path to my library contains a hidden file ($HOME/.local), so when the path is iterated through it skips over any directory starting with a period character. If I remove the first continue statement above, it seems to work fine.

Tim

**Comment by Axel-Naumann** [comment]
Wow, well done, Tim, thanks so much for your help with debugging this! I'll check what this code is meant to do.

**Comment by Axel-Naumann** [comment]
That's https://github.com/root-project/root/pull/9911 !

**Comment by Axel-Naumann** [comment]
Merged, thanks for reporting. Do you need this in the upcoming 6.26 or in 6.24, or is it enough to fix this in master and the future 6.28?

**Comment by belmakier** [comment]
It's not urgent, I can wait for 6.28.  Thanks!
